src/custom/pages/Profile/LockedGnoVesting/hooks.ts

```diff
@@ -0,0 +1,129 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { ContractTransaction } from '@ethersproject/contracts'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { useActiveWeb3React } from 'hooks/web3'
+import MERKLE_DROP_ABI from 'abis/MerkleDrop.json'
+import TOKEN_DISTRO_ABI from 'abis/TokenDistro.json'
+import { MerkleDrop, TokenDistro } from 'abis/types'
+import { useSingleCallResult } from 'lib/hooks/multicall'
+import { useTransactionAdder } from 'state/enhancedTransactions/hooks'
+import { useContract } from 'hooks/useContract'
+import { COW as COW_TOKENS } from 'constants/tokens'
+import { SupportedChainId } from 'constants/chains'
+import { OperationType } from 'components/TransactionConfirmationModal'
+import { fetchClaim } from './claimData'
+
+// We just generally use the mainnet version. We don't read from the contract anyways so the address doesn't matter
+const COW = COW_TOKENS[SupportedChainId.MAINNET]
+
+export const MERKLE_DROP_CONTRACT_ADDRESSES: Record<number, string> = {
+  [SupportedChainId.MAINNET]: '0x64646f112FfD6F1B7533359CFaAF7998F23C8c40',
+  [SupportedChainId.RINKEBY]: '0x5444c4AFb2ec7f7367C10F7732b8558650c5899F',
+  [SupportedChainId.XDAI]: '0x3d610e917130f9D036e85A030596807f57e11093',
+}
+
+const useMerkleDropContract = () => useContract<MerkleDrop>(MERKLE_DROP_CONTRACT_ADDRESSES, MERKLE_DROP_ABI, true)
+
+export const TOKEN_DISTRO_CONTRACT_ADDRESSES: Record<number, string> = {
+  [SupportedChainId.MAINNET]: '0x68FFAaC7A431f276fe73604C127Bd78E49070c92',
+  [SupportedChainId.RINKEBY]: '0xeBA8CE5b23c054f1511F8fF5114d848329B8258d',
+  [SupportedChainId.XDAI]: '0x3d610e917130f9D036e85A030596807f57e11093',
+}
+
+const useTokenDistroContract = () => useContract<TokenDistro>(TOKEN_DISTRO_CONTRACT_ADDRESSES, TOKEN_DISTRO_ABI, true)
```

Comment:
Not convinced we should, but what about moving these to the central place where similar constants/hooks are defined: `constants/tokens` and the hooks to [`hooks/useContract/index`](https://github.com/cowprotocol/cowswap/blob/develop/src/custom/hooks/useContract/index.ts#L33-L36)

Reply:
Updated